### ISSUE TITLE
Not able to get any rendered output

### ISSUE THREAD

**Issue by dilzeem** [pawamoy/mkdocstrings#140]
Hi I feel like I am missing something obvious here, but haven't been able to render anything. I also get no error messages. Just this: 
 

![image](https://user-images.githubusercontent.com/8899724/88382638-27419d00-cda9-11ea-874b-d2c45720bb34.png)

I expect this simple example should work, but it doesn't seem to be for some reason. Is there something I am missing here?

My folder structure is as follows:

```
.
├── docs
│   ├── index.md
│   └── sample-app.md
├── sample-app
│    └── sample_app.py
│    ├──  tests
│    │     └── __init__.py
│    │     └── some_test.py
│    ├── poetry.lock
│    └── pyproject.toml
│ 
└──  mkdocs.yml

```

and my `mkdocs.yml` is below 

```yaml
site_name: sample-app docs
theme:
  name: material

nav:
  - Code Docs: sample-app.md

plugins:
  - search:
  - mkdocstrings:
      default_handler: python
      handlers:
        python:
          rendering:
            show_source: true
      watch:
        - sample-app
```

The file `sample_app.py` contains the following.

```python
class MyClass:
    """Print print print!"""

    def method_a(self):
        """Print A!"""
        print("A!")

    def method_b(self):
        """Print B!"""
        print("B!")

    def method_c(self):
        """Print C!"""
        print("C!")
```

And finally the `sample-app.md` contains the following

```
# Documentation of the Code


::: sample-app.sample_app.MyClass

```

**Comment by pawamoy** [comment]
Hmmm, this is rather unusual folder structure (for what I saw in projects using mkdocstrings). Do you have a repo to share so I can clone and try it out :slightly_smiling_face: ?

**Comment by dilzeem** [comment]
Thanks for the quick reply.

https://github.com/dilzeem/onboarding

There is a little more in in there then what I have described above.

**Comment by pawamoy** [comment]
Found it!

- the regex for `::: identifier` instructions is not accepting `-`: we can consider adding it
- `importlib` fails to import `sample-app.sample_app`, probably because it's a namespace package

As a workaround, you could add the `sample-app` directory into your PYTHONPATH and use `::: sample_app.MyClass` instead.
It would be nice if we could also resolve point 2.

**Comment by pawamoy** [comment]
You can actually fix point 2 with this:

```yaml
plugins:
  - search:
  - mkdocstrings:
      default_handler: python
      handlers:
        python:
          setup_commands:
            - import sys
            - sys.path.append("sample-app")
          rendering:
            show_source: true
      watch:
        - sample-app
```

You Markdown files can then keep `::: sample-app.sample_app.MyClass`.

**Comment by dilzeem** [comment]
Hi @pawamoy, 

Thanks for the effort putting into debugging this. I don't think I would have picked up the `-` error.

I was able to resolve this correctly after some further bugging on my computer. I realized that I have `mkdocs` installed globally as well, and was using that to run `mkdoc serve` causing an import error.

I will close this issue as I will think about having a more conventional directory structure just make this process easier.

**Comment by dilzeem** [comment]
Hi @pawamoy, 

Thank you for putting the effort into debugging this. 

I was able to get it to work after some further debugging. My issue was that I was also using the global `mkdocs`  that I had installed, and not the one in my virtual environment. This was also causing an import error. 

Thank you once again for your assistance, and I will also look into adopting a more conventional directory structure.
